src/lib/utils/data-extraction.ts

```diff
@@ -34,16 +34,15 @@ export function getItemRects(list: HTMLUListElement): ItemRect[] {
 	);
 }
 
-export const getTextDirection = (element: HTMLElement): HTMLElement['dir'] => {
+export const getTextDirection = (element?: HTMLElement): HTMLElement['dir'] => {
```

Comment:
Even though there’s a check in the line below to make sure an `element` parameter is being passed, that’s just for extra security. This function shouldn’t even be called without `element` being passed.

```suggestion
export const getTextDirection = (element: HTMLElement): HTMLElement['dir'] => {
```